### PR TITLE
Initial commit of the bootstrap role

### PR DESCRIPTION
**Pull request by dasJ**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+On Arch, you also need the Ubuntu keyring package from the AUR.
+
+## Role Variables
+
+#### Global
+
+| Name               | Required           | Default | Description                                                                                   |
+|--------------------|:------------------:|---------|-----------------------------------------------------------------------------------------------|
+| `global_cache_dir` | :heavy_check_mark: |         | Path where files are downloaded and extracted. Also, the SD card is mounted in a subdirectory |
+| `global_cores`     | :heavy_check_mark: | `1`     | Amount of CPU cores in the control machine to build with                                      |
```
Comment (SlothOfAnarchy) [inline README.md]: why do you have a default if the var is mandatory? Same with the other vars below.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      bootstrap_gateway: 192.168.0.1
+```
+
+## License
+
+This work is licensed under a [Creative Commons Attribution-ShareAlike 4.0 International License](http://creativecommons.org/licenses/by-sa/4.0/).
+
+## Author Information
+
+- [Janne Heß](https://github.com/dasJ)
```
Comment (SlothOfAnarchy) [inline README.md:91]: please add your stuvus email

File: tasks/boot.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  copy:
+    content: "arm_control=0x200\nkernel=arm64-kernel.img\n"
+    dest: "{{ global_cache_dir }}/rpi3-mount/boot/config.txt"
+
+- name: Write kernel command line
+  copy:
+    content: "console=tty1 root=/dev/mmcblk1p2 rootfstype=ext4 rootwait rw init=/lib/systemd/systemd net.ifnames=0\n"
+    dest: "{{ global_cache_dir }}/rpi3-mount/boot/cmdline.txt"
+
+- name: Create fstab
```
Comment (SlothOfAnarchy) [inline tasks/boot.yml:49]: maybe extract the tasks that have nothing to do with /boot

File: tasks/debootstrap.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+---
+- name: Debootstrap Ubuntu
+  command:
+    cmd: >
+      debootstrap --foreign --arch arm64
+      --variant minbase
+      --include systemd,ifupdown,openssh-server,python,dnsutils,udev
+      "{{ bootstrap_ubuntu_release | mandatory }}"
+      "{{ global_cache_dir | mandatory }}/rpi3-mount"
+       http://ports.ubuntu.com/
```
Comment (SlothOfAnarchy) [inline tasks/debootstrap.yml:10]: is there no https version or a way to verify hashes/signatures?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+| `bootstrap_device`    | :heavy_check_mark: |                                | The SD card to install on (`/dev` path)       |
+| `bootstrap_qemu`      | :heavy_check_mark: | `/usr/bin/qemu-aarch64-static` | Path to the static aarch64 qemu on your sytem |
+| `bootstrap_boot_size` | :heavy_check_mark: | `1GiB`                         | Size of `/boot` on the Raspberry Pi           |
+
+#### Versions
+
+| Name                       | Required           | Default   | Description                                              |
+|----------------------------|:------------------:|-----------|----------------------------------------------------------|
+| `bootstrap_kernel_version` | :heavy_check_mark: | `4.11.y`  | Version of the Raspberry Pi kernel to build (git branch) |
+| `bootstrap_extraversion`   | :heavy_check_mark: | `+stuvus` | Extraversion to append on the kernel                     |
```
Comment (SlothOfAnarchy) [inline README.md]: why is the extra version mandatory?

File: tasks/mount.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+---
+- name: Mount root
+  mount:
+    src: "{{ bootstrap_device }}2"
+    path: "{{ global_cache_dir }}/rpi3-mount"
+    fstype: ext4
+    state: mounted
+    fstab: /tmp/fstab
+
+- name: Make boot directory
```
Comment (SlothOfAnarchy) [inline tasks/mount.yml]: "Create"

File: README.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+| `bootstrap_ubuntu_release` | :heavy_check_mark: | `yakkety` | Ubuntu release name to install                           |
+
+#### User
+
+| Name                      | Required                 | Default | Description                                        |
+|---------------------------|:------------------------:|---------|----------------------------------------------------|
+| `bootstrap_root_key`      | :heavy_multiplication_x: |         | Public SSH key to add to `authorized_keys` of root |
+| `bootstrap_root_password` | :heavy_multiplication_x: |         | Password to set for the root user                  |
```
Comment (SlothOfAnarchy) [inline README.md]: describe the kind of hash that is expected

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (dasJ) [reply README.md]: You could unset it, or is this impossible in Ansible?

File: README.md (hunk 5 above)
Comment (dasJ) [reply README.md]: You don't need it. The kernel will have a shorter version number, then

File: README.md (hunk 2 above)
Comment (dasJ) [reply README.md:91]: But I don't wish to receive mail

File: tasks/boot.yml (hunk 3 above)
Comment (dasJ) [reply tasks/boot.yml:49]: See comment in the first line. These tasks make the Pi bootable

File: tasks/debootstrap.yml (hunk 4 above)
Comment (dasJ) [reply tasks/debootstrap.yml:10]: Sadly, no. But the packages are verified against the Ubuntu keyring before extracting

**Comment by dasJ** [comment]
@SlothOfAnarchy Pls rereview

File: tasks/boot.yml (hunk 3 above)
Comment (SlothOfAnarchy) [reply tasks/boot.yml:49]: I don't see a comment in the first line?

File: README.md (hunk 1 above)
Comment (SlothOfAnarchy) [reply README.md]: Still a confusing approach to describe mandatory vars. Your :heavy_check_mark: leaves the impression that the user **has** to set it.

File: README.md (hunk 5 above)
Comment (SlothOfAnarchy) [reply README.md]: Don't mark it as mandatory then?

**Comment by SlothOfAnarchy** [comment]
please fix the required changes.

**Comment by dasJ** [comment]
@SlothOfAnarchy Pls rereview

File: tasks/boot.yml (hunk 3 above)
Comment (dasJ) [reply tasks/boot.yml:49]: Looks like I removed it

File: tasks/partitions.yml [hunk 8]
```diff
@@ -3,28 +3,35 @@
...
+    state: present
     part_end: "{{ bootstrap_boot_size | mandatory }}"
     part_type: primary
 
 - name: Create root partition
   parted:
     device: "{{ bootstrap_device | mandatory }}"
     number: 2
+    state: present
+    part_start: "{{ bootstrap_boot_size | mandatory }}"
```
Comment (SlothOfAnarchy) [inline tasks/partitions.yml]: why do you use the mandatory filter? `bootstrap_boot_size` is set by default.

File: tasks/partitions.yml [hunk 9]
```diff
@@ -3,28 +3,35 @@
...
+  changed_when: false
+  register: fatpart
+
 - name: Make FAT32
-  filesystem:
-    dev: "{{ bootstrap_device | mandatory }}1"
-    fstype: fat
-    opts: "-F 32 -n BOOT"
+  command: "mkfs.fat -F 32 -n BOOT {{ bootstrap_device | mandatory }}1"
+  when: "not 'ext4' in fatpart.stdout"
```
Comment (SlothOfAnarchy) [inline tasks/partitions.yml]: why do you only check against ext4?

Comment (dasJ) [reply tasks/partitions.yml]: The filesystem module does that. However, it does not support FAT

Comment (SlothOfAnarchy) [reply tasks/partitions.yml]: I meant your `when` condition

Comment (dasJ) [reply tasks/partitions.yml]: It was supposed to mean fat...

**Review by SlothOfAnarchy — APPROVED** [review]

**Pull request merged**